pandas/core/index.py

```diff
@@ -2543,11 +2550,12 @@ def insert(self, loc, item):
         -------
         new_index : Index
         """
-        _self = np.asarray(self)
-        item_idx = Index([item], dtype=self.dtype).values
-        idx = np.concatenate(
-            (_self[:loc], item_idx, _self[loc:]))
-        return Index(idx, name=self.name)
+        indexes=[self[:loc],
+                 Index([item]),
+                 self[loc:]]
+
+        return indexes[0].append(indexes[1]).append(indexes[2])
```

Comment:
`_shallow_copy` works great if the types are consistent. (`MultiIndex` could be solved separately).

But if they're not consistent, don't you need to recast the index? Or is this trying to solve for something we shouldn't be solving for?

```python
In [179]:
int_index=pd.Index(range(0,5))
int_index
Out[179]:
Int64Index([0, 1, 2, 3, 4], dtype='int64')

In [180]:
int_index._shallow_copy(np.append(int_index.values,"a"))
Out[180]:
Int64Index([u'0', u'1', u'2', u'3', u'4', u'a'], dtype='|S21')
```
...should that still be an `Int64Index`